src/app/pipes/timedisplay.ts

```diff
@@ -0,0 +1,45 @@
+import { PipeTransform, Pipe } from '@angular/core';
+
+
+@Pipe({name: 'timeDisplay'})
+export class TimeDisplayPipe implements PipeTransform {
+    transform(timeInSeconds: number, shortened = false): string {
+        let timeInSecondsLeft = timeInSeconds;
+        const days = Math.floor(timeInSecondsLeft / 86400);
+        timeInSecondsLeft = timeInSecondsLeft % 86400;
+        const hours = Math.floor(timeInSecondsLeft / 3600);
+        timeInSecondsLeft = timeInSecondsLeft % 3600;
+        const minutes = Math.floor(timeInSecondsLeft / 60);
+        timeInSecondsLeft = timeInSecondsLeft % 60;
+        const seconds = Math.floor(timeInSecondsLeft);
+
+        let time = '';
+        if (days > 0) {
+            time = this.transformHelper(time, days, 'days', shortened);
+        }
+        if (hours > 0) {
+            time = this.transformHelper(time, hours, 'hours', shortened);
+        }
+        if (minutes > 0) {
+            time = this.transformHelper(time, minutes, 'minutes', shortened);
+        }
+        if (days === 0 && seconds > 0) {
```

Comment:
I don't know if I like that, because it could potentially be misleading. For example, if one of the artifacts is stubbed out, it means that we have not been able to determine the time between them, not necessarily that the time is 0 s

Reply:
@sarah256 `null` is set if the wait time can't be determined, so in that case, it can just blank. If there is a 0, it should actually mean 0.